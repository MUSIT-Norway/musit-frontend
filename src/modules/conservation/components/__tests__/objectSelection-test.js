// @flow
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ObjectSelection from '../objectSelection';
import sinon from 'sinon';

describe('ObjectSelection', () => {
  const mainEvent = [
    {
      objectData: {
        id: 51,
        uuid: '12080e3e-2ca2-41b1-9d4a-4d72e292dcd8',
        museumId: 99,
        museumNo: 'MusN11',
        term: 'Solsikke',
        collection: 8,
        materials: [],
        locations: [],
        coordinates: [],
        objectType: 'collection'
      },
      id: 51,
      uuid: '12080e3e-2ca2-41b1-9d4a-4d72e292dcd8',
      museumId: 99,
      museumNo: 'MusN11',
      term: 'Solsikke',
      collection: 8,
      materials: [],
      locations: [],
      coordinates: [],
      objectType: 'collection'
    },
    {
      objectData: {
        id: 54,
        uuid: 'aba6a67c-f742-4a44-b13e-0415ec1abb2a',
        museumId: 99,
        museumNo: 'MusN13',
        term: 'Makrellsopp',
        collection: 6,
        materials: [],
        locations: [],
        coordinates: [],
        objectType: 'collection'
      },
      id: 54,
      uuid: 'aba6a67c-f742-4a44-b13e-0415ec1abb2a',
      museumId: 99,
      museumNo: 'MusN13',
      term: 'Makrellsopp',
      collection: 6,
      materials: [],
      locations: [],
      coordinates: [],
      objectType: 'collection'
    },
    {
      objectData: {
        id: 19,
        uuid: '788136fe-8038-47fc-9c59-9ddb010e0f74',
        museumId: 99,
        museumNo: 'MusN20',
        term: 'Ukjent mygg',
        collection: 2,
        materials: [],
        locations: [],
        coordinates: [],
        objectType: 'collection'
      },
      id: 19,
      uuid: '788136fe-8038-47fc-9c59-9ddb010e0f74',
      museumId: 99,
      museumNo: 'MusN20',
      term: 'Ukjent mygg',
      collection: 2,
      materials: [],
      locations: [],
      coordinates: [],
      objectType: 'collection'
    }
  ];
  const SubEventObjectUUIDs = [
    '12080e3e-2ca2-41b1-9d4a-4d72e292dcd8',
    'aba6a67c-f742-4a44-b13e-0415ec1abb2a'
  ];
  it('should render ObjectSelection for add and edit', () => {
    const preventDefault = sinon.spy();
    const updateFormCalled = sinon.spy();

    const wrapper = shallow(
      <ObjectSelection
        affectedThingsWithDetailsMainEvent={mainEvent}
        affectedThingsSubEvent={SubEventObjectUUIDs}
        viewMode={false}
        updateForm={updateFormCalled}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    wrapper
      .find('tr')
      .first()
      .simulate('click', { preventDefault });
    expect(updateFormCalled.calledOnce).toBe(false);
  });

  it('should render ObjectSelection as read only', () => {
    const wrapper = shallow(
      <ObjectSelection
        affectedThingsWithDetailsMainEvent={mainEvent}
        affectedThingsSubEvent={SubEventObjectUUIDs}
        viewMode={true}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
