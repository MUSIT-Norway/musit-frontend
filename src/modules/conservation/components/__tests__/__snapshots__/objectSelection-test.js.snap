// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectSelection should render ObjectSelection as read only 1`] = `
<div
  className="form-group"
>
  <div
    className="col-md-12 col-md-offset-0"
  >
    <div>
      <label>
        Hvilke objekter gjelder denne del hendelsen for?
      </label>
    </div>
    <table
      className="table table-striped table-responsive table-hover"
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <thead>
        <tr>
          <th>
            <input
              checked={false}
              className="normalAction"
              disabled={true}
              onChange={[Function]}
              type="checkbox"
            />
          </th>
          <th>
            Museumsnr.
          </th>
          <th>
            Undernr.
          </th>
          <th>
            Gjenstand/Takson
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          onClick={[Function]}
        >
          <td
            name="type"
            width={10}
          >
            <input
              checked={true}
              disabled={true}
              type="checkbox"
              value=""
            />
          </td>
          <td
            name="museumNo"
          >
            MusN11
          </td>
          <td
            name="subNo"
          />
          <td
            name="term"
          >
            Solsikke
          </td>
        </tr>
        <tr
          onClick={[Function]}
        >
          <td
            name="type"
            width={10}
          >
            <input
              checked={true}
              disabled={true}
              type="checkbox"
              value=""
            />
          </td>
          <td
            name="museumNo"
          >
            MusN13
          </td>
          <td
            name="subNo"
          />
          <td
            name="term"
          >
            Makrellsopp
          </td>
        </tr>
        <tr
          onClick={[Function]}
        >
          <td
            name="type"
            width={10}
          >
            <input
              checked={false}
              disabled={true}
              type="checkbox"
              value=""
            />
          </td>
          <td
            name="museumNo"
          >
            MusN20
          </td>
          <td
            name="subNo"
          />
          <td
            name="term"
          >
            Ukjent mygg
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`ObjectSelection should render ObjectSelection for add and edit 1`] = `
<div
  className="form-group"
>
  <div
    className="col-md-12 col-md-offset-0"
  >
    <div>
      <label>
        Hvilke objekter gjelder denne del hendelsen for?
      </label>
    </div>
    <table
      className="table table-striped table-responsive table-hover"
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <thead>
        <tr>
          <th>
            <input
              checked={false}
              className="normalAction"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
          </th>
          <th>
            Museumsnr.
          </th>
          <th>
            Undernr.
          </th>
          <th>
            Gjenstand/Takson
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          onClick={[Function]}
        >
          <td
            name="type"
            width={10}
          >
            <input
              checked={true}
              disabled={false}
              type="checkbox"
              value=""
            />
          </td>
          <td
            name="museumNo"
          >
            MusN11
          </td>
          <td
            name="subNo"
          />
          <td
            name="term"
          >
            Solsikke
          </td>
        </tr>
        <tr
          onClick={[Function]}
        >
          <td
            name="type"
            width={10}
          >
            <input
              checked={true}
              disabled={false}
              type="checkbox"
              value=""
            />
          </td>
          <td
            name="museumNo"
          >
            MusN13
          </td>
          <td
            name="subNo"
          />
          <td
            name="term"
          >
            Makrellsopp
          </td>
        </tr>
        <tr
          onClick={[Function]}
        >
          <td
            name="type"
            width={10}
          >
            <input
              checked={false}
              disabled={false}
              type="checkbox"
              value=""
            />
          </td>
          <td
            name="museumNo"
          >
            MusN20
          </td>
          <td
            name="subNo"
          />
          <td
            name="term"
          >
            Ukjent mygg
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
